src/highlight/HighlightListener.ts

```diff
@@ -0,0 +1,92 @@
+import debounce from 'lodash/debounce';
+import { AppStore, getIsInitialized, SelectionArg, setSelectionAction } from '../store';
+import { MOUSE_PRIMARY } from '../constants';
+
+export type Options = {
+    getSelection: () => SelectionArg | null;
```

Comment:
nit: feels a bit funny that the return type has name `Arg` in it, could `Selection` type be used here?

Reply:
The `Selection` type is defined in docUtil, which I want to separate with highlight. I can rename `SelectionArg` to `Selection` in the import.